dev/deno.ts

```diff
@@ -600,11 +620,142 @@ const denoDoc: Fig.Subcommand = {
+            }
+          } catch (err) {
+            console.error("Returning early due to error:", err);
+            return [];
+          }
+
+          const showPrivateSymbols = tokens.includes("--private");
+
+          if (docNodes.length === 0) return [];
+
+          // The final token has to be the filter, it's the only way this
+          // generator could have been invoked.
+          const filterToken = tokens[tokens.length - 1];
+          const firstDotIndex = filterToken.indexOf(".");
+
+          // Based on whether the user has typed a period, this will be
+          // populated with either the top level nodes or the children of
+          // whatever node the user has searched for.
+          const suggestNodes = [];
+
+          if (firstDotIndex === -1) {
+            suggestNodes.push(...docNodes);
+          } else {
+            const filterName = filterToken.slice(0, firstDotIndex);
+
+            // It's not uncommon that there'd be multiple occurrences of the
+            // same name with different values, eg. overloads and interface
+            // merging. Deno's builtin types actually do this with the `Deno`
+            // interface. Typically this will only be one or two nodes.
+            const foundNodes = docNodes.filter((node) => {
+              return node.name === filterName;
+            });
+
+            // `deno doc` can only generate docs for these nodes' children
+            for (const node of foundNodes) {
+              if (node.kind === "namespace") {
+                suggestNodes.push(...node.namespaceDef.elements);
+              } else if (node.kind === "interface") {
+                suggestNodes.push(
+                  ...node.interfaceDef.methods,
+                  ...node.interfaceDef.properties
+                );
+              } else if (node.kind === "class") {
+                suggestNodes.push(
+                  ...node.classDef.methods,
+                  ...node.classDef.properties
+                );
+              }
+            }
+
+            // It's a common case to have no children, so it's worth checking.
+            if (suggestNodes.length === 0) return [];
+          }
+
+          // The names are added to a set because duplicates are common.
+          const suggestionNames = new Set<string>();
+
+          // Which nodes are visible is dependent on --private
+          if (showPrivateSymbols) {
+            for (const node of suggestNodes) {
+              suggestionNames.add(node.name);
+            }
+          } else {
+            for (const node of suggestNodes) {
+              // Deno includes imports in the JSON regardless of privacy,
+              // probably so it's easier to build a module graph. However,
+              // imports are (by definition) private, so they must be removed.
+              if (node.kind === "import") continue;
+              suggestionNames.add(node.name);
+            }
+          }
```

Comment:
We can avoid duplicating the `for` loop to make this a bit cleaner:
```suggestion
          if (showPrivateSymbols) {
          for (const node of suggestNodes) {
            // Deno includes imports in the JSON regardless of privacy,
            // probably so it's easier to build a module graph. However,
            // imports are (by definition) private, so they must be removed.
            if (!showPrivateSymbols && node.kind === "import") continue;
            suggestionNames.add(node.name);
          }
```

Reply:
It's cleaner, but it ends up checking the `showPrivateSymbols` on each iteration instead of just once. That's probably an inconsequential difference, but it's a bunch of work that _can_ be skipped. If you think the trade off is worth it I'm happy to implement!

Reply:
How many `node` can be contained inside `suggestNodes`? I don't think that an `if` would cause any issues even if there are thousands of nodes, and I found to not repeat the `for` loop is cleaner. What do you think of it?